pkg/danmep/interface.go

```diff
@@ -0,0 +1,73 @@
+package danmep
+
+import (
+    "fmt"
+
+    danmtypes "github.com/nokia/danm/crd/apis/danm/v1"
+    danmclientset "github.com/nokia/danm/crd/client/clientset/versioned"
+    "github.com/nokia/danm/pkg/danmep"
+    "github.com/nokia/danm/pkg/ipam"
+)
+
+// GetSupportedReleaseIPHandlers gets all supported static CNI implementation that cleaner can use
+// when deleting dangling DanmEps
+func GetSupportedReleaseIPHandlers() []*ReleaseIPService {
+    return []*ReleaseIPService{
+        NewReleaseIPService(&calicoReleaseIPServiceImpl{}),
+    }
+}
+
+// ReleaseIPInterface need to be implemented by every static CNI IPAM release plugin
+// that want to invoked when Cleaner is cleaning up dangling DanmEps
+type ReleaseIPInterface interface {
+    ReleaseIP(dnet *danmtypes.DanmNet, ep *danmtypes.DanmEp, ip string) error
+    IsIPAllocatedByMe(dnet *danmtypes.DanmNet, ep *danmtypes.DanmEp, ip string) bool
+}
+
+type ReleaseIPService struct {
+    handler ReleaseIPInterface
+}
+
+// NewReleaseIPService constructor encapsulates different ReleaseIPInterface implementations under a generic API
+func NewReleaseIPService(handler ReleaseIPInterface) *ReleaseIPService {
+    return &ReleaseIPService{handler}
+}
+
+// IsIPAllocatedByMe invokes embedded implementation specific IsIPAllocatedByMe function
+func (d ReleaseIPService) IsIPAllocatedByMe(dnet *danmtypes.DanmNet, ep *danmtypes.DanmEp, ip string) bool {
+    return ! ipam.WasIpAllocatedByDanm(ip, dnet.Spec.Options.Cidr) && d.handler.IsIPAllocatedByMe(dnet, ep, ip)
```

Comment:
Spec.Options.Cidr is always the V4 CIDR
the V6 IP should be cehcked against the V6 allocation CIDR